lib/payoneer/client.rb

```diff
@@ -66,17 +66,19 @@ def expanded_payout(payee_id:, client_reference_id:, amount:, currency:, descrip
       }
 
       encoded_credentials = 'Basic ' + Base64.encode64("#{configuration.username}:#{configuration.api_password}").chomp
-      response = RestClient.post "#{configuration.json_base_uri}/payouts", params.to_json, content_type: 'application/json', accept: :json, Authorization: encoded_credentials
-      raise ResponseError.new(code: response.code, body: response.body) if response.code != 200
 
-      hash = JSON.parse(response.body)
-      hash['PaymentID'] = hash['payout_id'] # Keep consistent with the normal payout response body
+      begin
+        response = RestClient.post "#{configuration.json_base_uri}/payouts", params.to_json, content_type: 'application/json', accept: :json, Authorization: encoded_credentials
 
-      if hash.key?('Code')
-        Response.new(hash['Code'], hash['Description'])
-      else
-        hash = block_given? ? yield(hash) : hash
-        Response.new(Response::OK_STATUS_CODE, hash)
+        hash = JSON.parse(response.body)
+        hash['PaymentID'] = hash['payout_id'] # Keep consistent with the normal payout response body
+
+        create_response(hash)
+      rescue RestClient::Exception => e
+        if e.http_body
+          hash = JSON.parse(e.http_body)
+          create_response(hash)
```

Comment:
We should pass the status code down as well, otherwise we might accidentally report a 200 when we didn't mean to.